Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -87,6 +87,8 @@ def fixed_name(file, bad_name)
       "@loader_path/#{bad_name}"
     elsif file.mach_o_executable? && (lib + bad_name).exist?
       "#{lib}/#{bad_name}"
+    elsif bad_name.start_with? "@rpath"
+      bad_name.sub("@rpath", lib)
```

Comment:
The `LC_RPATH` may be set to `#{lib}`, but it could be set to something else. You'll need to emulate the logic of the dynamic linker and search each directory specified by `LC_RPATH` to locate the given library, to determine the correct absolute path.